### ISSUE TITLE
tablets: creating a keyspace with tablets enabled and without NTS succeeds and results in using vnodes

### ISSUE THREAD

**Issue by patjed41** [scylladb/scylladb#19743]
In the Data Distribution with Tablets doc we write "The recommended `NetworkTopologyStrategy` for keyspaces remains *required* when using tablets."

However, with tablets enabled, the following CQL statements succeed without even logging a warning:
`CREATE KEYSPACE ks WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 3} AND tablets = {'enabled': true};`
`CREATE KEYSPACE ks WITH replication = {'class': 'EverywhereStrategy', 'replication_factor': 3} AND tablets = {'enabled': true};`

A keyspace created with one of the statements above works by using vnodes. This is very strange and confusing. We should fail such CQL statements instead.

**Comment by kbr-scylla** [comment]
Assigning to frontend team

cc @kostja @bhalevy

**Comment by ptrsmrn** [comment]
Possible duplicate of https://github.com/scylladb/scylladb/issues/19744

**Comment by kbr-scylla** [comment]
>  The above example is also wrong for another reason, mainly - SimpleStrategy is not supported for tablets.

So there are two different reasons, this issue is about the SimpleStrategy one, the other one is about `enable_tablets` in Scylla config. I'd say two different issues are appropriate. Doesn't mean they cannot be solved with single PR

**Comment by ptrsmrn** [comment]
@kbr-scylla I agree, to me it seems like some validation was (accidentally?) turned OFF for tablets, which results in different outcomes, but *maybe* the root cause is the same.

**Comment by mykaul** [comment]
@ptrsmrn - this is going to be VERY confusing for users - do we plan to fix it soon?

**Comment by ptrsmrn** [comment]
@mykaul we plan to start working on the fix this/next week.

**Comment by xemul** [comment]
Fixed by #20779
